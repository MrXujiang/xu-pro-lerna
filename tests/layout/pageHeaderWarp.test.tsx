import { render, mount } from 'enzyme';
import React from 'react';
import ProLayout, { PageContainer } from '@ant-design/pro-layout';
import { act } from 'react-dom/test-utils';
import defaultProps from './defaultProps';
import { waitForComponentToPaint } from '../util';

describe('BasicLayout', () => {
  it('base use', () => {
    const html = render(
      <ProLayout {...defaultProps}>
        <PageContainer />
      </ProLayout>,
    );
    expect(html).toMatchSnapshot();
  });

  it('content is text', () => {
    const html = render(
      <ProLayout {...defaultProps}>
        <PageContainer content="just so so" />
      </ProLayout>,
    );
    expect(html).toMatchSnapshot();
  });

  it('title=false, don not render title view', async () => {
    const wrapper = mount(
      <ProLayout {...defaultProps}>
        <PageContainer title={false} />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    expect(wrapper.find('.ant-page-header-heading-title')).toHaveLength(0);
  });

  it('have default title', async () => {
    const wrapper = mount(
      <ProLayout {...defaultProps}>
        <PageContainer />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    const titleDom = wrapper.find('.ant-page-header-heading-title');
    expect(titleDom.text()).toEqual('welcome');
  });

  it('title overrides the default title', async () => {
    const wrapper = mount(
      <ProLayout {...defaultProps}>
        <PageContainer title="name" />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    const titleDom = wrapper.find('.ant-page-header-heading-title');
    expect(titleDom.text()).toEqual('name');
  });

  it('with default prefixCls props TopNavHeader', async () => {
    const wrapper = mount(
      <ProLayout
        {...defaultProps}
        layout="mix"
        splitMenus
        isMobile={false}
        rightContentRender={() => <span />}
      >
        <PageContainer title="name" />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    const domHeader = wrapper.find('.ant-pro-top-nav-header-logo');

    act(() => {
      wrapper.setProps({
        rightContentRender: () => (
          <div
            style={{
              width: 200,
            }}
          >
            xx
          </div>
        ),
      });
    });
    expect(domHeader.exists()).toBe(true);
  });

  it('without custom prefixCls props TopNavHeader', async () => {
    const prefixCls = 'ant-oh-pro';
    const wrapper = mount(
      <ProLayout {...defaultProps} layout="top" prefixCls={prefixCls}>
        <PageContainer title="name" />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    const domHeader = wrapper.find(`.${prefixCls}-top-nav-header-logo`);
    expect(domHeader.exists()).toBe(true);
  });

  it('pageHeaderRender return false', async () => {
    const wrapper = mount(
      <ProLayout {...defaultProps} layout="top">
        <PageContainer title="name" pageHeaderRender={() => null} />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    const domHeader = wrapper.find('ant-page-header');
    expect(domHeader.exists()).toBeFalsy();
    act(() => {
      wrapper.unmount();
    });
  });

  it('pageHeaderRender is false', async () => {
    const wrapper = mount(
      <ProLayout {...defaultProps} layout="top">
        <PageContainer title="name" pageHeaderRender={false} />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    const domHeader = wrapper.find('ant-page-header');
    expect(domHeader.exists()).toBeFalsy();
    act(() => {
      wrapper.unmount();
    });
  });
});
