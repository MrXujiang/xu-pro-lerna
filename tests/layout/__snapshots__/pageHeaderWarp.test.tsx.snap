// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLayout base use 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout ant-pro-basicLayout-side"
>
  <section
    class="ant-layout"
    style="min-height:100%"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark ant-pro-sider ant-pro-sider-layout-side"
      style="overflow:hidden;flex:0 0 208px;max-width:208px;min-width:208px;width:208px"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-logo"
          id="logo"
        >
          <a>
            <svg
              height="32px"
              viewBox="0 0 200 200"
              width="32px"
            >
              <defs>
                <linearGradient
                  id="linearGradient-1"
                  x1="62.1023273%"
                  x2="108.19718%"
                  y1="0%"
                  y2="37.8635764%"
                >
                  <stop
                    offset="0%"
                    stop-color="#4285EB"
                  />
                  <stop
                    offset="100%"
                    stop-color="#2EC7FF"
                  />
                </linearGradient>
                <linearGradient
                  id="linearGradient-2"
                  x1="69.644116%"
                  x2="54.0428975%"
                  y1="0%"
                  y2="108.456714%"
                >
                  <stop
                    offset="0%"
                    stop-color="#29CDFF"
                  />
                  <stop
                    offset="37.8600687%"
                    stop-color="#148EFF"
                  />
                  <stop
                    offset="100%"
                    stop-color="#0A60FF"
                  />
                </linearGradient>
                <linearGradient
                  id="linearGradient-3"
                  x1="69.6908165%"
                  x2="16.7228981%"
                  y1="-12.9743587%"
                  y2="117.391248%"
                >
                  <stop
                    offset="0%"
                    stop-color="#FA816E"
                  />
                  <stop
                    offset="41.472606%"
                    stop-color="#F74A5C"
                  />
                  <stop
                    offset="100%"
                    stop-color="#F51D2C"
                  />
                </linearGradient>
                <linearGradient
                  id="linearGradient-4"
                  x1="68.1279872%"
                  x2="30.4400914%"
                  y1="-35.6905737%"
                  y2="114.942679%"
                >
                  <stop
                    offset="0%"
                    stop-color="#FA8E7D"
                  />
                  <stop
                    offset="51.2635191%"
                    stop-color="#F74A5C"
                  />
                  <stop
                    offset="100%"
                    stop-color="#F51D2C"
                  />
                </linearGradient>
              </defs>
              <g
                fill="none"
                fill-rule="evenodd"
                stroke="none"
                stroke-width="1"
              >
                <g
                  transform="translate(-20.000000, -20.000000)"
                >
                  <g
                    transform="translate(20.000000, 20.000000)"
                  >
                    <g>
                      <g
                        fill-rule="nonzero"
                      >
                        <g>
                          <path
                            d="M91.5880863,4.17652823 L4.17996544,91.5127728 C-0.519240605,96.2081146 -0.519240605,103.791885 4.17996544,108.487227 L91.5880863,195.823472 C96.2872923,200.518814 103.877304,200.518814 108.57651,195.823472 L145.225487,159.204632 C149.433969,154.999611 149.433969,148.181924 145.225487,143.976903 C141.017005,139.771881 134.193707,139.771881 129.985225,143.976903 L102.20193,171.737352 C101.032305,172.906015 99.2571609,172.906015 98.0875359,171.737352 L28.285908,101.993122 C27.1162831,100.824459 27.1162831,99.050775 28.285908,97.8821118 L98.0875359,28.1378823 C99.2571609,26.9692191 101.032305,26.9692191 102.20193,28.1378823 L129.985225,55.8983314 C134.193707,60.1033528 141.017005,60.1033528 145.225487,55.8983314 C149.433969,51.69331 149.433969,44.8756232 145.225487,40.6706018 L108.58055,4.05574592 C103.862049,-0.537986846 96.2692618,-0.500797906 91.5880863,4.17652823 Z"
                            fill="url(#linearGradient-1)"
                          />
                          <path
                            d="M91.5880863,4.17652823 L4.17996544,91.5127728 C-0.519240605,96.2081146 -0.519240605,103.791885 4.17996544,108.487227 L91.5880863,195.823472 C96.2872923,200.518814 103.877304,200.518814 108.57651,195.823472 L145.225487,159.204632 C149.433969,154.999611 149.433969,148.181924 145.225487,143.976903 C141.017005,139.771881 134.193707,139.771881 129.985225,143.976903 L102.20193,171.737352 C101.032305,172.906015 99.2571609,172.906015 98.0875359,171.737352 L28.285908,101.993122 C27.1162831,100.824459 27.1162831,99.050775 28.285908,97.8821118 L98.0875359,28.1378823 C100.999864,25.6271836 105.751642,20.541824 112.729652,19.3524487 C117.915585,18.4685261 123.585219,20.4140239 129.738554,25.1889424 C125.624663,21.0784292 118.571995,14.0340304 108.58055,4.05574592 C103.862049,-0.537986846 96.2692618,-0.500797906 91.5880863,4.17652823 Z"
                            fill="url(#linearGradient-2)"
                          />
                        </g>
                        <path
                          d="M153.685633,135.854579 C157.894115,140.0596 164.717412,140.0596 168.925894,135.854579 L195.959977,108.842726 C200.659183,104.147384 200.659183,96.5636133 195.960527,91.8688194 L168.690777,64.7181159 C164.472332,60.5180858 157.646868,60.5241425 153.435895,64.7316526 C149.227413,68.936674 149.227413,75.7543607 153.435895,79.9593821 L171.854035,98.3623765 C173.02366,99.5310396 173.02366,101.304724 171.854035,102.473387 L153.685633,120.626849 C149.47715,124.83187 149.47715,131.649557 153.685633,135.854579 Z"
                          fill="url(#linearGradient-3)"
                        />
                      </g>
                      <ellipse
                        cx="100.519339"
                        cy="100.436681"
                        fill="url(#linearGradient-4)"
                        rx="23.6001926"
                        ry="23.580786"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </svg>
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
        <div
          style="flex:1;overflow-y:auto;overflow-x:hidden"
        >
          <ul
            class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark ant-pro-sider-menu"
            data-menu-list="true"
            role="menu"
            style="width:100%"
            tabindex="0"
          >
            <li
              class="ant-menu-submenu ant-menu-submenu-inline"
              role="none"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                class="ant-menu-submenu-title"
                role="menuitem"
                style="padding-left:16px"
                tabindex="-1"
              >
                <span
                  class="ant-menu-title-content"
                >
                  <span
                    class="ant-pro-menu-item"
                    title="welcome"
                  >
                    welcome
                  </span>
                </span>
                <i
                  class="ant-menu-submenu-arrow"
                />
              </div>
            </li>
          </ul>
          <div
            aria-hidden="true"
            style="display:none"
          />
        </div>
        <div
          class="ant-pro-sider-links"
        >
          <ul
            class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark ant-pro-sider-link-menu"
            data-menu-list="true"
            role="menu"
            tabindex="0"
          >
            <li
              class="ant-menu-item ant-menu-item-only-child ant-pro-sider-collapsed-button"
              role="menuitem"
              style="padding-left:16px"
              tabindex="-1"
            >
              <span
                class="ant-menu-title-content"
              >
                <span
                  aria-label="menu-fold"
                  class="anticon anticon-menu-fold"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="menu-fold"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                    />
                  </svg>
                </span>
              </span>
            </li>
          </ul>
          <div
            aria-hidden="true"
            style="display:none"
          />
        </div>
      </div>
    </aside>
    <div
      class="ant-layout"
      style="position:relative"
    >
      <header
        class="ant-layout-header ant-pro-fixed-header-action ant-pro-header-dark"
        style="padding:0;height:48px;line-height:48px;width:100%;z-index:19"
      >
        <div
          class="ant-pro-global-header ant-pro-global-header-layout-side"
        >
          <div
            style="flex:1"
          />
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-page-container"
        >
          <div
            class="ant-pro-page-container-warp"
          >
            <div
              class="ant-page-header"
            >
              <div
                class="ant-page-header-heading"
              >
                <div
                  class="ant-page-header-heading-left"
                >
                  <span
                    class="ant-page-header-heading-title"
                    title="welcome"
                  >
                    welcome
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
</div>
`;

exports[`BasicLayout content is text 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout ant-pro-basicLayout-side"
>
  <section
    class="ant-layout"
    style="min-height:100%"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark ant-pro-sider ant-pro-sider-layout-side"
      style="overflow:hidden;flex:0 0 208px;max-width:208px;min-width:208px;width:208px"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-logo"
          id="logo"
        >
          <a>
            <svg
              height="32px"
              viewBox="0 0 200 200"
              width="32px"
            >
              <defs>
                <linearGradient
                  id="linearGradient-1"
                  x1="62.1023273%"
                  x2="108.19718%"
                  y1="0%"
                  y2="37.8635764%"
                >
                  <stop
                    offset="0%"
                    stop-color="#4285EB"
                  />
                  <stop
                    offset="100%"
                    stop-color="#2EC7FF"
                  />
                </linearGradient>
                <linearGradient
                  id="linearGradient-2"
                  x1="69.644116%"
                  x2="54.0428975%"
                  y1="0%"
                  y2="108.456714%"
                >
                  <stop
                    offset="0%"
                    stop-color="#29CDFF"
                  />
                  <stop
                    offset="37.8600687%"
                    stop-color="#148EFF"
                  />
                  <stop
                    offset="100%"
                    stop-color="#0A60FF"
                  />
                </linearGradient>
                <linearGradient
                  id="linearGradient-3"
                  x1="69.6908165%"
                  x2="16.7228981%"
                  y1="-12.9743587%"
                  y2="117.391248%"
                >
                  <stop
                    offset="0%"
                    stop-color="#FA816E"
                  />
                  <stop
                    offset="41.472606%"
                    stop-color="#F74A5C"
                  />
                  <stop
                    offset="100%"
                    stop-color="#F51D2C"
                  />
                </linearGradient>
                <linearGradient
                  id="linearGradient-4"
                  x1="68.1279872%"
                  x2="30.4400914%"
                  y1="-35.6905737%"
                  y2="114.942679%"
                >
                  <stop
                    offset="0%"
                    stop-color="#FA8E7D"
                  />
                  <stop
                    offset="51.2635191%"
                    stop-color="#F74A5C"
                  />
                  <stop
                    offset="100%"
                    stop-color="#F51D2C"
                  />
                </linearGradient>
              </defs>
              <g
                fill="none"
                fill-rule="evenodd"
                stroke="none"
                stroke-width="1"
              >
                <g
                  transform="translate(-20.000000, -20.000000)"
                >
                  <g
                    transform="translate(20.000000, 20.000000)"
                  >
                    <g>
                      <g
                        fill-rule="nonzero"
                      >
                        <g>
                          <path
                            d="M91.5880863,4.17652823 L4.17996544,91.5127728 C-0.519240605,96.2081146 -0.519240605,103.791885 4.17996544,108.487227 L91.5880863,195.823472 C96.2872923,200.518814 103.877304,200.518814 108.57651,195.823472 L145.225487,159.204632 C149.433969,154.999611 149.433969,148.181924 145.225487,143.976903 C141.017005,139.771881 134.193707,139.771881 129.985225,143.976903 L102.20193,171.737352 C101.032305,172.906015 99.2571609,172.906015 98.0875359,171.737352 L28.285908,101.993122 C27.1162831,100.824459 27.1162831,99.050775 28.285908,97.8821118 L98.0875359,28.1378823 C99.2571609,26.9692191 101.032305,26.9692191 102.20193,28.1378823 L129.985225,55.8983314 C134.193707,60.1033528 141.017005,60.1033528 145.225487,55.8983314 C149.433969,51.69331 149.433969,44.8756232 145.225487,40.6706018 L108.58055,4.05574592 C103.862049,-0.537986846 96.2692618,-0.500797906 91.5880863,4.17652823 Z"
                            fill="url(#linearGradient-1)"
                          />
                          <path
                            d="M91.5880863,4.17652823 L4.17996544,91.5127728 C-0.519240605,96.2081146 -0.519240605,103.791885 4.17996544,108.487227 L91.5880863,195.823472 C96.2872923,200.518814 103.877304,200.518814 108.57651,195.823472 L145.225487,159.204632 C149.433969,154.999611 149.433969,148.181924 145.225487,143.976903 C141.017005,139.771881 134.193707,139.771881 129.985225,143.976903 L102.20193,171.737352 C101.032305,172.906015 99.2571609,172.906015 98.0875359,171.737352 L28.285908,101.993122 C27.1162831,100.824459 27.1162831,99.050775 28.285908,97.8821118 L98.0875359,28.1378823 C100.999864,25.6271836 105.751642,20.541824 112.729652,19.3524487 C117.915585,18.4685261 123.585219,20.4140239 129.738554,25.1889424 C125.624663,21.0784292 118.571995,14.0340304 108.58055,4.05574592 C103.862049,-0.537986846 96.2692618,-0.500797906 91.5880863,4.17652823 Z"
                            fill="url(#linearGradient-2)"
                          />
                        </g>
                        <path
                          d="M153.685633,135.854579 C157.894115,140.0596 164.717412,140.0596 168.925894,135.854579 L195.959977,108.842726 C200.659183,104.147384 200.659183,96.5636133 195.960527,91.8688194 L168.690777,64.7181159 C164.472332,60.5180858 157.646868,60.5241425 153.435895,64.7316526 C149.227413,68.936674 149.227413,75.7543607 153.435895,79.9593821 L171.854035,98.3623765 C173.02366,99.5310396 173.02366,101.304724 171.854035,102.473387 L153.685633,120.626849 C149.47715,124.83187 149.47715,131.649557 153.685633,135.854579 Z"
                          fill="url(#linearGradient-3)"
                        />
                      </g>
                      <ellipse
                        cx="100.519339"
                        cy="100.436681"
                        fill="url(#linearGradient-4)"
                        rx="23.6001926"
                        ry="23.580786"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </svg>
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
        <div
          style="flex:1;overflow-y:auto;overflow-x:hidden"
        >
          <ul
            class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark ant-pro-sider-menu"
            data-menu-list="true"
            role="menu"
            style="width:100%"
            tabindex="0"
          >
            <li
              class="ant-menu-submenu ant-menu-submenu-inline"
              role="none"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                class="ant-menu-submenu-title"
                role="menuitem"
                style="padding-left:16px"
                tabindex="-1"
              >
                <span
                  class="ant-menu-title-content"
                >
                  <span
                    class="ant-pro-menu-item"
                    title="welcome"
                  >
                    welcome
                  </span>
                </span>
                <i
                  class="ant-menu-submenu-arrow"
                />
              </div>
            </li>
          </ul>
          <div
            aria-hidden="true"
            style="display:none"
          />
        </div>
        <div
          class="ant-pro-sider-links"
        >
          <ul
            class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark ant-pro-sider-link-menu"
            data-menu-list="true"
            role="menu"
            tabindex="0"
          >
            <li
              class="ant-menu-item ant-menu-item-only-child ant-pro-sider-collapsed-button"
              role="menuitem"
              style="padding-left:16px"
              tabindex="-1"
            >
              <span
                class="ant-menu-title-content"
              >
                <span
                  aria-label="menu-fold"
                  class="anticon anticon-menu-fold"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="menu-fold"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                    />
                  </svg>
                </span>
              </span>
            </li>
          </ul>
          <div
            aria-hidden="true"
            style="display:none"
          />
        </div>
      </div>
    </aside>
    <div
      class="ant-layout"
      style="position:relative"
    >
      <header
        class="ant-layout-header ant-pro-fixed-header-action ant-pro-header-dark"
        style="padding:0;height:48px;line-height:48px;width:100%;z-index:19"
      >
        <div
          class="ant-pro-global-header ant-pro-global-header-layout-side"
        >
          <div
            style="flex:1"
          />
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-page-container"
        >
          <div
            class="ant-pro-page-container-warp"
          >
            <div
              class="ant-page-header"
            >
              <div
                class="ant-page-header-heading"
              >
                <div
                  class="ant-page-header-heading-left"
                >
                  <span
                    class="ant-page-header-heading-title"
                    title="welcome"
                  >
                    welcome
                  </span>
                </div>
              </div>
              <div
                class="ant-page-header-content"
              >
                <div
                  class="ant-pro-page-container-detail"
                >
                  <div
                    class="ant-pro-page-container-main"
                  >
                    <div
                      class="ant-pro-page-container-row"
                    >
                      <div
                        class="ant-pro-page-container-content"
                      >
                        just so so
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
</div>
`;
